addons/stock_account/models/stock.py

```diff
@@ -19,15 +19,33 @@ class StockInventory(models.Model):
         help="Date at which the accounting entries will be created"
              " in case of automated inventory valuation."
              " If empty, the inventory date will be used.")
+    has_account_moves = fields.Boolean(
+        default=False, store=False,
+        compute='_compute_has_account_moves')
+
+    def _compute_has_account_moves(self):
+        for inventory in self:
+            inventory.has_account_moves = False
+            if inventory.state == 'done':
+                account_move = self.env['account.move'].search([('stock_move_id.id', 'in', inventory.move_ids.ids)])
+                if account_move.exists():
+                    inventory.has_account_moves = True
+
+    def action_get_account_moves(self):
+        self.ensure_one()
+        action_ref = self.env.ref('account.action_move_journal_line')
+        if not action_ref:
+            return False
+        action_data = action_ref.read()[0]
+        action_data['domain'] = [('stock_move_id.id', 'in', self.move_ids.ids)]
+        return action_data
 
     @api.multi
     def post_inventory(self):
-        acc_inventories = self.filtered(lambda inventory: inventory.accounting_date)
-        for inventory in acc_inventories:
+        for inventory in self:
+            if not inventory.accounting_date:
+                inventory.accounting_date = fields.Date.today()
             super(StockInventory, inventory.with_context(force_period_date=inventory.accounting_date)).post_inventory()
-        other_inventories = self - acc_inventories
-        if other_inventories:
-            super(StockInventory, other_inventories).post_inventory()
```

Comment:
you change the logic a bit much here, is it wanted?

Reply:
It's wanted, but I think I misunderstood this part and the old version of this chunk of code is better than mine. I'll get back this chunk, thanks.